<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from kuka_iiwa_ctrl.xacro           | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="kuka_iiwa">
  <material name="Black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="Blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="Green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.2 0.2 0.2 1.0"/>
  </material>
  <material name="Orange">
    <color rgba="1.0 0.4235294117647059 0.0392156862745098 1.0"/>
  </material>
  <material name="Brown">
    <color rgba="0.8705882352941177 0.8117647058823529 0.7647058823529411 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <link name="world"/>
  <!--joint between {parent} and link_0-->
  <joint name="world_iiwa_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="iiwa_link_0"/>
  </joint>
  <link name="iiwa_link_0">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.1 0 0.07"/>
      <mass value="5"/>
      <inertia ixx="0.05" ixy="0" ixz="0" iyy="0.06" iyz="0" izz="0.03"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/visual/link_0.stl"/>
      </geometry>
      <material name="Grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/collision/link_0.stl"/>
      </geometry>
      <material name="Grey"/>
    </collision>
    <self_collision_checking>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <capsule length="0.25" radius="0.15"/>
      </geometry>
    </self_collision_checking>
  </link>
  <!-- joint between link_0 and link_1 -->
  <joint name="iiwa_joint_1" type="revolute">
    <parent link="iiwa_link_0"/>
    <child link="iiwa_link_1"/>
    <origin rpy="0 0 0" xyz="0 0 0.15"/>
    <axis xyz="0 0 1"/>
    <limit effort="300" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="10"/>
  </joint>
  <link name="iiwa_link_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0 -0.03 0.12"/>
      <mass value="3.4525"/>
      <inertia ixx="0.02183" ixy="0" ixz="0" iyy="0.007703" iyz="-0.003887" izz="0.02083"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0075"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/visual/link_1.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0075"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/collision/link_1.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="iiwa_joint_2" type="revolute">
    <parent link="iiwa_link_1"/>
    <child link="iiwa_link_2"/>
    <origin rpy="1.5707963267948966   0 3.141592653589793" xyz="0 0 0.19"/>
    <axis xyz="0 0 1"/>
    <limit effort="300" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="10"/>
    <safety_controller k_position="100" k_velocity="2" soft_lower_limit="-2.0594885173533086" soft_upper_limit="2.0594885173533086"/>
    <dynamics damping="0.5"/>
  </joint>
  <link name="iiwa_link_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0003 0.059 0.042"/>
      <mass value="3.4821"/>
      <inertia ixx="0.02076" ixy="0" ixz="-0.003626" iyy="0.02179" iyz="0" izz="0.00779"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/visual/link_2.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/collision/link_2.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="iiwa_joint_3" type="revolute">
    <parent link="iiwa_link_2"/>
    <child link="iiwa_link_3"/>
    <origin rpy="1.5707963267948966 0 3.141592653589793" xyz="0 0.21 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="300" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="10"/>
  </joint>
  <link name="iiwa_link_3">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.03 0.13"/>
      <mass value="4.05623"/>
      <inertia ixx="0.03204" ixy="0" ixz="0" iyy="0.00972" iyz="0.006227" izz="0.03042"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.026"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/visual/link_3.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.026"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/collision/link_3.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="iiwa_joint_4" type="revolute">
    <parent link="iiwa_link_3"/>
    <child link="iiwa_link_4"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 0 0.19"/>
    <axis xyz="0 0 1"/>
    <limit effort="300" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="10"/>
  </joint>
  <link name="iiwa_link_4">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.067 0.034"/>
      <mass value="3.4822"/>
      <inertia ixx="0.02178" ixy="0" ixz="0" iyy="0.02075" iyz="-0.003625" izz="0.007785"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/visual/link_4.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/collision/link_4.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="iiwa_joint_5" type="revolute">
    <parent link="iiwa_link_4"/>
    <child link="iiwa_link_5"/>
    <origin rpy="-1.5707963267948966 3.141592653589793 0" xyz="0 0.21 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="300" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="10"/>
  </joint>
  <link name="iiwa_link_5">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0001 0.021 0.076"/>
      <mass value="2.1633"/>
      <inertia ixx="0.01287" ixy="0" ixz="0" iyy="0.005708" iyz="-0.003946" izz="0.01112"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.026"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/visual/link_5.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.026"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/collision/link_5.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="iiwa_joint_6" type="revolute">
    <parent link="iiwa_link_5"/>
    <child link="iiwa_link_6"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 0.06070 0.19"/>
    <axis xyz="0 0 1"/>
    <limit effort="300" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="10"/>
  </joint>
  <link name="iiwa_link_6">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.0006 0.0004"/>
      <mass value="2.3466"/>
      <inertia ixx="0.006509" ixy="0" ixz="0" iyy="0.006259" iyz="0.00031891" izz="0.004527"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/visual/link_6.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/collision/link_6.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <joint name="iiwa_joint_7" type="revolute">
    <parent link="iiwa_link_6"/>
    <child link="iiwa_link_7"/>
    <origin rpy="-1.5707963267948966 3.141592653589793 0" xyz="0 0.081 0.06070"/>
    <axis xyz="0 0 1"/>
    <limit effort="300" lower="-3.0543261909900763" upper="3.0543261909900763" velocity="10"/>
  </joint>
  <link name="iiwa_link_7">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.02"/>
      <mass value="3.129"/>
      <inertia ixx="0.01464" ixy="0.0005912" ixz="0" iyy="0.01465" iyz="0" izz="0.002872"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.0005"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/visual/link_7.stl"/>
      </geometry>
      <material name="Grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.0005"/>
      <geometry>
        <mesh filename="package://kuka_iiwa_support/meshes/iiwa7/collision/link_7.stl"/>
      </geometry>
      <material name="Grey"/>
    </collision>
  </link>
  <joint name="camera_joint" type="fixed">
    <parent link="iiwa_link_7"/>
    <child link="kuka_iiwa/camera_link"/>
    <origin rpy="0 -1.57 0" xyz="0 0 0.16"/>
  </joint>
  <link name="kuka_iiwa/camera_link"/>
  <joint name="iiwa_joint_ee" type="fixed">
    <parent link="iiwa_link_7"/>
    <child link="iiwa_link_ee"/>
    <origin rpy="0 0 0" xyz="0 0 0.045"/>
  </joint>
  <link name="iiwa_link_ee">
    </link>
  <transmission name="iiwa_joint_1_tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="iiwa_joint_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="iiwa_joint_1_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="iiwa_joint_2_tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="iiwa_joint_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="iiwa_joint_2_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="iiwa_joint_3_tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="iiwa_joint_3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="iiwa_joint_3_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="iiwa_joint_4_tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="iiwa_joint_4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="iiwa_joint_4_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="iiwa_joint_5_tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="iiwa_joint_5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="iiwa_joint_5_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="iiwa_joint_6_tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="iiwa_joint_6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="iiwa_joint_6_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="iiwa_joint_7_tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="iiwa_joint_7">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="iiwa_joint_7_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>kuka_iiwa</robotNamespace>
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
      <legacyModeNS>true</legacyModeNS>
    </plugin>
  </gazebo>
  <gazebo reference="iiwa_link_0">
    <material>Gazebo/Black</material>
  </gazebo>
  <gazebo reference="iiwa_link_1">
    <material>Gazebo/Orange</material>
  </gazebo>
  <gazebo reference="iiwa_link_2">
    <material>Gazebo/Grey</material>
  </gazebo>
  <gazebo reference="iiwa_link_3">
    <material>Gazebo/Orange</material>
  </gazebo>
  <gazebo reference="iiwa_link_4">
    <material>Gazebo/Grey</material>
  </gazebo>
  <gazebo reference="iiwa_link_5">
    <material>Gazebo/Orange</material>
  </gazebo>
  <gazebo reference="iiwa_link_6">
    <material>Gazebo/Grey</material>
  </gazebo>
  <!-- CAMERA SENSOR -->
  <gazebo reference="kuka_iiwa/camera_link">
    <sensor name="camera" type="camera">
      <update_rate>30.0</update_rate>
      <camera>
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>800</width>
          <height>600</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>30</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <mean>0.0</mean>
          <stddev>0.007</stddev>
        </noise>
      </camera>
      <plugin filename="libgazebo_ros_camera.so" name="camera_controller">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>camera</cameraName>
        <imageTopicName>/kuka_iiwa/camera/image_raw</imageTopicName>
        <cameraInfoTopicName>/kuka_iiwa/camera/camera_info</cameraInfoTopicName>
        <frameName>kuka_iiwa/camera_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
    </sensor>
  </gazebo>
</robot>

